src/app/rpmostree-builtin-kargs.c

```diff
@@ -0,0 +1,203 @@
+                                       option_entries,
+                                       &argc, &argv,
+                                       invocation,
+                                       cancellable,
+                                       NULL, NULL,
+                                       &sysroot_proxy,
+                                       &peer_pid,
+                                       error))
+    return EXIT_FAILURE;
+
+  /* TODO: uncomment these lines when finish implementing editor feature
+   *
+   * if (opt_editor && (opt_import_proc || opt_kernel_delete_strings ||
+   *   opt_kernel_replace_strings ||  opt_kernel_append_strings))
+   *  {
+   *    We want editor command to achieve all the other functionalities
+   *   Thus erroring out ahead of time when multiple options exist
+   *   g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+   *                "Cannot specify --editor with other options");
+   *   return EXIT_FAILURE;
+   * }
+   */
+
+  if (opt_kernel_delete_strings && opt_kernel_replace_strings)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot specify both --delete and --replace");
+      return EXIT_FAILURE;
+    }
+  if (opt_kernel_delete_strings && opt_kernel_append_strings)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot specify both --delete and --append");
+      return EXIT_FAILURE;
+    }
+  if (opt_import_proc_cmdline && opt_deployid)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Canno specify both --import-from-proc-cmdline and --deployid");
+      return EXIT_FAILURE;
+    }
+  if (opt_import_proc_cmdline && opt_osname)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot specify both osname and --import-from-proc-cmdline");
+      return EXIT_FAILURE;
+    }
+  if (!(opt_kernel_delete_strings) && !(opt_kernel_append_strings)
+      && !(opt_kernel_replace_strings))
+    display_kernel_args = TRUE;
+
+  if (opt_reboot && display_kernel_args)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_INVALID_ARGUMENT,
+                   "Cannot reboot when kernel arguments not changed");
+      return EXIT_FAILURE;
+    }
+
+  glnx_unref_object RPMOSTreeOS *os_proxy = NULL;
+  if (!rpmostree_load_os_proxy (sysroot_proxy, opt_osname,
+                                cancellable, &os_proxy, error))
+    return EXIT_FAILURE;
+
+  /* The proc cmdline is the kernel args from booted deployment
+   * if this option is not specified, we will default to find the first
+   * pending  deployment that matches the osname if there is one
+   */
+  gboolean is_pending = !opt_import_proc_cmdline;
+
+  const char* deploy_str = opt_deployid ?: "";
+  g_autoptr(GVariant) boot_config = NULL;
+  if (!rpmostree_os_call_get_deployment_boot_config_sync (os_proxy,
```

Comment:
Just noticed via a bit more testing, the policy kit verification will be called twice since there are two different os methods being called. Will do a bit more further investigation on how to avoid that from happening.